python/cudf/cudf/core/column/numerical.py

```diff
@@ -105,8 +107,14 @@ def _apply_scan_op(self, op):
     def normalize_binop_value(self, other):
         if other is None:
             return other
+        if isinstance(other, cudf.Scalar):
+            # expensive device-host transfer just to
+            # adjust the dtype
+            other = other.value
```

Comment:
This particular set of typecasting rules needs a more thought out solution here. The plan currently is to handle this in a follow up.

Reply:
Would suggest adding a `#TODO` comment